src/components/MyReservation.js

```diff
@@ -44,12 +45,28 @@ const MyReservation = () => {
             <div key={reservation.id} className="flex items-center justify-center w-11/12 mt-5 ">
               <div className="flex flex-col items-center justify-around w-screen py-5 bg-white rounded-lg shadow-lg sm:flex-row">
                 <div className="w-20">
-                  <img className="rounded-full" src={doctor.data[reservation.doctor_id - 1].photo} alt="doctor" />
+                  <img
+                    className="rounded-full"
+                    src={doctor.data.map((doc) => {
+                      if (doc.id === reservation.doctor_id) {
+                        return doc.photo;
+                      }
+                      return '';
+                    })}
+                    alt="doctor"
+                  />
```

Comment:
This fix will ensure that the index correctly represents the desired doctor and return the proper picture. Great fix! 💯